edx_exams/apps/core/api.py

```diff
@@ -86,33 +90,51 @@ def _allow_status_transition(attempt_obj, to_status):
     """
     Helper method to assert that a given status transition is allowed
     """
-    allowed = True
-    error_message = ''
-
     # check that status is legal
     is_transition_legal = ExamAttemptStatus.is_status_transition_legal(attempt_obj.status, to_status)
-    allowed = allowed and is_transition_legal
+    if not is_transition_legal:
+        illegal_status_transition_msg = (
+            f'A status transition from "{attempt_obj.status}" to "{to_status}" was attempted '
+            f'on exam_id={attempt_obj.exam.id} for user_id={attempt_obj.user.id}. This is not '
+            f"allowed! (course_id={attempt_obj.exam.course_id})"
+        )
+        return False, illegal_status_transition_msg
+    return True, ''
 
-    illegal_status_transition_msg = (
-        f'A status transition from "{attempt_obj.status}" to "{to_status}" was attempted '
-        f'on exam_id={attempt_obj.exam.id} for user_id={attempt_obj.user.id}. This is not '
-        f"allowed! (course_id={attempt_obj.exam.course_id})"
-    )
-    error_message = error_message if is_transition_legal else illegal_status_transition_msg
-
-    # check that exam is allowed to start
-    if to_status == ExamAttemptStatus.started:
-        is_start_allowed = not (attempt_obj.status == ExamAttemptStatus.started and attempt_obj.start_time)
-        allowed = allowed and is_start_allowed
 
+def _check_exam_is_allowed_to_start(attempt_obj, user_id):
+    """
+    Helper method to assert if an exam is allowed to start
+    """
+    # Check the attempt not already "started" and no start time exists
+    exam_already_started = attempt_obj.status == ExamAttemptStatus.started and attempt_obj.start_time
+    if exam_already_started:
         illegal_start_msg = (
             f'Cannot start exam attempt for exam_id={attempt_obj.exam.id} '
             f'and user_id={attempt_obj.user.id} because it has already started!'
         )
+        return False, illegal_start_msg
 
-        error_message = error_message if is_start_allowed else illegal_start_msg
+    # Check that the user is not trying to go from `ready_to_submit` to `started`
+    ready_to_submit = attempt_obj.status == ExamAttemptStatus.ready_to_submit
```

Comment:
This actually should be allowed. When a learner clicks "Ready to Submit" but then decides to go back to the exam, the status transitions from `ready_to_submit` to `started`.